src/finch/compiled.py

```diff
@@ -50,6 +50,13 @@ def wrapper_func(*args, **kwargs):
             )
             if tag is not None:
                 compute_kwargs["tag"] = tag
+
+            if isinstance(result, Iterable):
+                computed = jl.Finch.compute(
+                    *map(lambda x: x._obj, result), **compute_kwargs
+                )
+                return tuple(Tensor(c) for c in computed)
+
```

Comment:
We can remove all of this and replace the line below by:

```python
return _recurse(result, lambda x: Tensor(jl.Finch.compute(x._obj, **compute_kwargs)))
```

Reply:
I disagree here - the point is to call `compute` once with all lazy tensors that come out of the same compiled function at the same time.
In your line we're calling `compute` multiple times. One time for each output.

Reply:
Hmm. I'll push to your branch with a fix for both issues. 😄

Reply:
https://github.com/finch-tensor/finch-tensor-python/pull/94/commits/28cf1b088a8d50787fde6288655bc851fb3b1066